modules/graceful/manager.go

```diff
@@ -217,15 +213,6 @@ func (g *Manager) IsChild() bool {
 // The order of closure is IsShutdown, IsHammer (potentially), IsTerminate
 func (g *Manager) IsShutdown() <-chan struct{} {
 	g.lock.RLock()
-	if g.shutdown == nil {
-		g.lock.RUnlock()
-		g.lock.Lock()
-		if g.shutdown == nil {
-			g.shutdown = make(chan struct{})
-		}
-		defer g.lock.Unlock()
-		return g.shutdown
-	}
 	defer g.lock.RUnlock()
 	return g.shutdown
```

Comment:
Please forgive me if I'm too dense, but... is it RLock() even needed here? I mean, g.shutdown is a stable struct now (it will not be replaced by another channel for the duration of the manager), and a channel is inherently safe to read in multithreading... 🤔

Reply:
Yup this is what you get for causing me to completely restructure complex synchronization, it was safe, if unnecessarily so...